editor/js/SkinnedMuscle.js

```diff
@@ -48,14 +65,52 @@ THREE.SkinnedMuscle.prototype.updateMatrixWorld = function( force ) {
     var mat = new THREE.Matrix4().getInverse(this.parent.matrixWorld);
     var vec = new THREE.Vector3().setFromMatrixPosition(mat);
 
-    for (var b = 0; b < this.pathpoints.length; b++) {
-        var nextPathpointObject = this.pathpointObjects[b];
-        if (nextPathpointObject !== undefined) {
-            this.children[b].position.setFromMatrixPosition(nextPathpointObject.matrixWorld);
-            this.children[b].position.add(vec);
-            //bones[b].pos.setFromMatrixPosition(pptObject.matrixWorld);
-            //console.warn('bone '+b+' pos ='+bones[b].position.x, bones[b].position.y, bones[b].position.z);
-            this.children[b].updateMatrixWorld();
+    // Variables for the two points of a given path segement, the axis to
+    // be rotated (from) and the vector between them (to)
+    var pt1 = new THREE.Vector3();
+    var pt2 = new THREE.Vector3();
+    var vFrom = new THREE.Vector3(0, -1, 0);
+    var vTo = new THREE.Vector3();
+
+    // cycle through each segement defined by two PathPoints, pt1 and pt2
+    // and alignge the bones (caps of each segment) to be alinged with
+    // the vector connecting them.
+    var b = 0; // bone (of SkinnedMuscle) index
+    for (var p = 0; p < this.pathpoints.length-1; p++) {
+        var thisPathpointObject = this.pathpointObjects[p];
+        var nextPathpointObject = this.pathpointObjects[p+1];
+
+        if(thisPathpointObject !== undefined) {
+            pt1.setFromMatrixPosition(thisPathpointObject.matrixWorld);
+            pt2.setFromMatrixPosition(nextPathpointObject.matrixWorld);
+
+            vTo = pt2.clone();
+            vTo.sub(pt1).normalize();
+
+            // bones are positioned on the pathpoints
+            bones[b].position.setFromMatrixPosition(thisPathpointObject.matrixWorld);
+            bones[b].position.add(vec);
+            // the orientation of the bone is updated to have its Y-axis pointed
+            // back along the vector from pt1 to pt2
+            bones[b].quaternion.setFromUnitVectors(vFrom, vTo);
+
+            bones[++b].position.setFromMatrixPosition(nextPathpointObject.matrixWorld);
+            // the orientation of the bone is updated to have its Y-axis pointed
+            // back along the vector from pt1 to pt2
+            bones[b].position.add(vec);
+            bones[b].quaternion.setFromUnitVectors(vFrom, vTo);
+
+            if (this.DEBUG) {
+                console.warn("Path point name: " + thisPathpointObject.name);
+                console.warn("Num pathpoints = " + this.pathpoints.length);
+                console.warn("Num bones = " + this.children.length);
+                console.warn("This is vec:" + vec.toArray());
+                //console.dir(this.children[2*p+1]);
+                this.DEBUG = false;
+            }
+
+            bones[b-1].updateMatrixWorld();
+            bones[b++].updateMatrixWorld();
         }
```

Comment:
I think the Skeleton update call already calls the bone(s) updateMatrixWorld. can we check if this is necessary? Thanks @aseth1